newlib/libc/stdio/nano-vfprintf.c

```diff
@@ -229,7 +230,7 @@ _DEFUN(__ssputs_r, (ptr, fp, buf, len),
   if (len < w)
     w = len;
 
-  (void)memmove ((_PTR) fp->_p, (_PTR) buf, (size_t) (w));
+  (void)memcpy_P ((_PTR) fp->_p, (_PTR) buf, (size_t) (w));
```

Comment:
Where is `memcpy_P` defined?

Reply:
It's not, and is linked from the Arduino SDK libs.  Cut-n-pasting that version somewhere in newlib's not difficult, but I imagine it's not good code hygiene.

Reply:
How about moving the definition into newlib then (adding at here and removing from Arduino)? As mentioned below, I would like libc to stay more or less independent of the application which is built on top of it, if this is reasonably easy to do. In this case, i think it is.